docs/user/AUTH.md

```diff
@@ -0,0 +1,42 @@
+# Kerberos authentication
```

Comment:
I think we should move this entire section to a Confluence page, it's not relevant to non-Red Hatters

Reply:
I am not entirely sure on this. True that it is Red Hat specific like eg. errata collector etc. At least for now. It is however defined here in the repo so the implementation is not internal. I would not bother much by moving it anywhere now but in the future we could either move it to some Red Hat internal place or generalize/parametrize it so it can be theoretically used by someone else too and offer it as one of the auth methods. For now I would extend the AUTH.md documentation with the local instance (the default) method.

Reply:
I don't mean the entire documentation, just the AUTH part, most of the things explained here give context into RH's internal kerberos implementation (which I will have to remove anyway) and only apply to RH, even if someone else were to use our kerberos + ldap module they most likely wouldn't encounter all the issues with `dns_canonicalize_hostname`, the different realms, etc.

Reply:
As of the chat discussion the Kerberos related chapter will be moved to `krb5_auth` and preserved there for now.